src/python/pants/engine/target.py

```diff
@@ -2466,10 +2473,35 @@ def unevaluated_transitive_excludes(self) -> UnparsedAddressInputs:
         )
 
 
+# If this predicate returns false, then the target's field's deps will be ignored.
+ShouldTraverseDepsPredicate = Callable[[Target, Field], bool]
+
+
+# noinspection PyUnusedLocal
+def should_traverse_deps_default_predicate(tgt: Target, fld: Field) -> bool:
```

Comment:
Consider renaming these to match their behavior, and not so much match the expected usage-scenario.
e.g. `traverse_if_dependencies_field` and `always_traverse` (below)

Reply:
This is because the usage is hard to parse:
```python
should_traverse_deps_predicate=should_traverse_all_deps_predicate,
```
vs:
```python
should_traverse_deps_predicate=always_traverse,
```